CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -245,6 +245,9 @@ class CRM_Contribute_Form_AbstractEditPayment extends CRM_Contact_Form_Task {
    */
   public function preProcess() {
     $this->_contactID = CRM_Utils_Request::retrieve('cid', 'Positive', $this);
+    if (empty($this->_contactID)) {
+      $this->_contactID = CRM_Utils_Request::retrieve('contact_id', 'Positive');
+    }
```

Comment:
If we've submitted the form, preprocess is retriggered so get the contactID from the submission and the rest of the preprocess function will fill the form object with the necessary parameters.